### PR TITLE
feat: custom http client

### PR DESCRIPTION
**Pull request by Vinzent03**
# Motivation
Same reason as https://github.com/supabase-community/postgrest-dart/pull/66
# Implementation
Instead of passing the `httpClient` down to the api, I replaced the global `fetch`, because tests use the global variable to mock `Fetch`. I think this pattern is very uncommon, but I didn't want to rewrite all tests.

### CODE REVIEW

**Comment by dshukertjr** [comment]
@Vinzent03 Thank you so much. Sorry, the one we merged is a pretty major one.

**Comment by Vinzent03** [comment]
@dshukertjr Yeah I know. I saw it some weeks ago, but it was quite inactive. Of course, you merge it just before this one 😉

**Comment by Vinzent03** [comment]
@dshukertjr May you review this pr?

**Comment by dshukertjr** [comment]
@Vinzent03 So sorry for taking my time to hop onto this one. I will take a look right now.

File: lib/src/types.dart [hunk 1]
```diff
@@ -189,7 +189,7 @@ class StorageException {
   const StorageException(this.message, {this.error, this.statusCode});
 
   StorageException.fromJson(Map<String, dynamic> json, [String? statusCode])
-      : message = json['message'] as String,
+      : message = json['message'] as String? ?? json.toString(),
```
Comment (dshukertjr) [inline lib/src/types.dart:192]: Love this little touch 😁

**Review by dshukertjr — APPROVED** [review]
Thanks for another amazing PR! 

One note: we will be shipping this PR along with [this](https://github.com/supabase-community/storage-dart/pull/30) PR, which includes a breaking change, and other packages needs similar update, so it might take a week or two until this change is shipped.

**Comment by Vinzent03** [comment]
@dshukertjr Thanks for merging. No problem with delayed release!

**Pull request merged**